skbio/io/_base.py

```diff
@@ -185,3 +186,25 @@ def _format_fasta_like_records(generator, id_whitespace_replacement,
                 "scores associated with it." % cardinal_to_ordinal(idx + 1))
 
         yield header, seq.sequence, seq.quality
+
+
+def _line_generator(fh, skip_blanks=False):
+    for line in fh:
+        line = line.strip()
+        if line or not skip_blanks:
+            yield line
+
+
+def _too_many_blanks(fh, max_blanks):
+    count = 0
+    too_many = False
+    for line in _line_generator(fh):
+        if line:
+            break
+        else:
+            count += 1
+            if count > max_blanks:
+                too_many = True
+                break
+    fh.seek(0)
```

Comment:
I like that idea, but I don't think it will work because the sniffers don't handle the lines directly; they invoke `_fasta_to_generator` and `_fastq_to_generator` with the file handle. We could add a parameter to the generators if we really want to avoid `seek`.

Barring that my recommendation would be checking if the user has `seek`ed to an offset, saving the offset, sniffing, then resetting to the user's value.